src/viam/sdk/robot/service.cpp

```diff
@@ -42,8 +42,8 @@ RobotService_::RobotService_(const std::shared_ptr<ResourceManager>& manager, Se
 std::vector<ResourceName> RobotService_::generate_metadata_() {
     std::vector<ResourceName> metadata;
     for (const auto& key_and_val : resource_manager()->resources()) {
-        for (const Name& resource : resource_names_for_resource(key_and_val.second)) {
-            metadata.push_back(resource.to_proto());
+        for (const Name& name : key_and_val.second->registered_models()) {
```

Comment:
If this is really the only usage of `registered_models`, I'd suggest just moving the function into this file in the unnamed namespace and changing it to take a `const Resource&`